abci/proposals/proposals.go

```diff
@@ -276,6 +270,19 @@ func (h *ProposalHandler) ProcessProposalHandler() sdk.ProcessProposalHandler {
 			req.Txs = req.Txs[NumInjectedTxs:]
 		}
 
-		return h.processProposalHandler(ctx, req)
+		// record time spent in slinky-specific ProcesssProposal
+		processDuration := time.Since(start)
+
+		// call the wrapped process-proposal
+		resp, err := h.processProposalHandler(ctx, req)
+		if err == nil {
+			// record time spent in slinky-specific ProcesssProposal on successful process-proposals
+			h.logger.Info(
+				"recording process proposal time",
+				"duration (seconds)", processDuration.Seconds(),
+			)
+			h.metrics.ObserveABCIMethodLatency(servicemetrics.ProcessProposal, processDuration)
+		}
+		return resp, err
```

Comment:
The reason I had metrics observation in a defer was to guarantee that we reported a metric no matter what our response was.

The way you have it above will only report in the happy path and not any other result. I think we want to report metrics no matter whether the response is Accept or Reject.

Reply:
Ya, I intentionally wanted to only collect latencies for successful requests.

Reply:
You'd prefer to report in all cases?